### ISSUE TITLE
Error displayed in Godot on exiting Rider

### ISSUE THREAD

**Issue by paulloz** [JetBrains/godot-support#133]
- Godot: v4.2.stable.mono.official [46dc27791]
- Rider: Build #RD-233.14015.60, built on January 18, 2024
- Windows: 10.0.19045 N/A build 19045

The following expression is printed in Godot when closing Rider. It happens reliably every time. I am not entirely sure if this is due to Rider not closing the connection properly, or Godot mishandling the disconnection request.

```
 /root/godot/modules/mono/editor/GodotTools/GodotTools/Ides/GodotIdeManager.cs:229 - Unhandled exception in the peer loop
  System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..
   ---> System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.
     --- End of inner exception stack trace ---
     at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
     at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
     at System.IO.StreamReader.ReadBufferAsync(CancellationToken cancellationToken)
     at System.IO.StreamReader.ReadLineAsyncInternal(CancellationToken cancellationToken)
     at GodotTools.IdeMessaging.Peer.ReadLine() in /root/godot/modules/mono/editor/GodotTools/GodotTools.IdeMessaging/Peer.cs:line 186
     at GodotTools.IdeMessaging.Peer.Process() in /root/godot/modules/mono/editor/GodotTools/GodotTools.IdeMessaging/Peer.cs:line 91
```

**Comment by van800** [comment]
I have tried to reproduce with 4.2.stable.mono.official, but unfortunately it doesn't happen for me.
I have a fix https://github.com/JetBrains/godot-support/commit/94e6e1a77c5f3b58e368f04ea5afe2eef4dda5fe, but can't test it.

**Comment by paulloz** [comment]
I'll try to install the whole Gradle stack to test it out as soon as I find some time 🙏

**Comment by van800** [comment]
I will make a plugin build for you.

**Comment by van800** [comment]
[rider-godot-2024.1.163.zip](https://github.com/JetBrains/godot-support/files/15049132/rider-godot-2024.1.163.zip)
Install it in Rider 2024.1.1 from Settings-> Plugins-> cog icon ->Install plugin from Disk
<img width="1018" alt="image" src="https://github.com/JetBrains/godot-support/assets/1482681/79f1d324-b86a-415e-8b32-20dbf83e0870">

**Comment by paulloz** [comment]
Thank you! I can confirm this fixes the issue entirely 👍

**Comment by van800** [comment]
released in version 2024.1.164
